core/src/main/java/org/apache/iceberg/V2Metadata.java

```diff
@@ -560,6 +561,11 @@ public F copy() {
       throw new UnsupportedOperationException("Cannot copy IndexedDataFile wrapper");
     }
 
+    @Override
+    public F copyWithStats(Set<Integer> columnsToKeepStats) {
```

Comment:
I think it is debatable whether to use `String` or `Integer` here. Clearly, `Integer` is easier but why should the API expose column IDs? I suspect that this should be `String` unless there is something that makes it far more difficult.

Reply:
Is there an easy way to convert column name to column id in case of schema evolution? How should we handle cases where the column is renamed, or a column is dropped and a new column is created with the same name?
Would it be enough to use the `Schema.findField` for this, and accept that we can only enable the feature for current columns?